src/node_crypto.cc

```diff
@@ -5735,6 +5759,71 @@ void RandomBytes(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void RandomBytesBuffer(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!args[0]->IsUint8Array()) {
+    return env->ThrowTypeError("buf must be a Buffer");
+  }
+
+  const int64_t len = Buffer::Length(args[0]);
+
+  if (!args[1]->IsUint32()) {
+    return env->ThrowTypeError("offset must be a number >= 0");
+  }
+
+  if (!args[2]->IsUint32()) {
+    return env->ThrowTypeError("size must be a number >= 0");
+  }
+
+  int64_t offset = args[1]->IntegerValue();
+  int64_t size = args[2]->IntegerValue();
+
+  if (offset < 0 || offset > Buffer::kMaxLength || offset > len)
+    return env->ThrowRangeError("offset out of range");
+
+  if (size < 0 || size > Buffer::kMaxLength)
+    return env->ThrowRangeError("size out of range");
+
+  if (size + offset > len)
+    return env->ThrowRangeError("buffer too small");
```

Comment:
Is there a reason for doing the type and length checks in C++ land?  You could do them in JS land and simply CHECK them here.